### PR TITLE
Miscellaneous tweaks

### PR DESCRIPTION
**Pull request by FrenchOrange**
*Removed the unused IDs for the Schoolboy class. The file "Parties.asm" has no corresponding data for those, and it looks like neither do the final games.

*Changed "BUG_CATCHER_BENNY" to just "BENNY" for the sake of uniformisation.
 
*Changed the first two IDs for the Super Nerd class, so that they match "Parties.asm".

*Added a mention to a seemingly unused trainer (initial encounter - Super Nerd Eric).

### CODE REVIEW

File: constants/trainer_constants.asm [hunk 1]
```diff
@@ -414,8 +408,8 @@ NUM_NONTRAINER_PHONECONTACTS EQU const_value - 1
 	const HUEY3
 
 	trainerclass SUPER_NERD ; 29
-	const STAN
-	const ERIC
+	const ERIC1 ; unused
+	const ERIC2
```
Comment (Rangi42) [inline constants/trainer_constants.asm:412]: `ERIC2` is used in GoldenrodUnderground; please change these to `ERIC_UNUSED` and `ERIC`, like Crystal's `NAOKO_UNUSED` and `NAOKO`.

**Comment by mid-kid** [comment]
Thanks a lot for your contribution! The suggestions seem fine, though I'd keep `BUG_CATCHER_BENNY` for uniformity with crystal - that has a `BIKER_BENNY` as well. In any circumstance the constants names should reflect the strings in the ROM.

**Comment by Rangi42** [comment]
I'm fine with the change to `BENNY`. There are other changes made to omit knowledge from Crystal, like renaming some `GSFuncName` functions to just `FuncName`, or `ENGINE_SWARM` vs `ENGINE_FISH_SWARM`.

**Comment by Rangi42** [comment]
This does not build: the code still uses the old `BUG_CATCHER_BENNY` name.

**Review by Rangi42 — CHANGES_REQUESTED** [review]
`Unknown symbol "SuperNerdEricUnusedBattleText"`

Make sure that `make compare` works.

**Comment by vulcandth** [comment]
@FrenchOrange It looks like your making your changes directly on github.com. The ideal way to update a PR is to checkout your branch [FrenchOrange:patch-1 ](https://github.com/FrenchOrange/pokegold/tree/patch-1) on your local computer. Do your changes, run `make compare` to verify that your changes build successfully, and then finally you git push a commit with your changes to your local branch. The PR will automatically update with any changes made to your local branch that your are attempting to pull into the main repo.

**Review by Rangi42 — APPROVED** [review]

**Pull request merged**